service/src/main/kotlin/fi/espoo/evaka/pis/service/MergeService.kt

```diff
@@ -84,9 +84,13 @@ class MergeService(private val asyncJobRunner: AsyncJobRunner<AsyncJob>) {
             UPDATE message_content SET author_id = (SELECT id FROM message_account WHERE person_id = :id_master) WHERE author_id = (SELECT id FROM message_account WHERE person_id = :id_duplicate);
             UPDATE message_recipients SET recipient_id = (SELECT id FROM message_account WHERE person_id = :id_master) WHERE recipient_id = (SELECT id FROM message_account WHERE person_id = :id_duplicate);
             UPDATE message_draft SET account_id = (SELECT id FROM message_account WHERE person_id = :id_master) WHERE account_id = (SELECT id FROM message_account WHERE person_id = :id_duplicate);
+            
+            UPDATE message_thread_participant mtp SET folder_id = (SELECT id FROM message_thread_folder WHERE owner_id = mtp.participant_id AND name = 'ARCHIVE') WHERE mtp.folder_id = (SELECT id FROM message_thread_folder WHERE owner_id = mtp.participant_id AND name = 'ARCHIVE');
+            DELETE FROM message_thread_folder WHERE owner_id = (SELECT id FROM message_account WHERE person_id = :id_master) AND name = 'ARCHIVE';
+            UPDATE message_thread_folder SET owner_id = (SELECT id FROM message_account WHERE person_id = :id_master) WHERE owner_id = (SELECT id FROM message_account WHERE person_id = :id_duplicate);
```

Comment:
These queries do this:
- The first `UPDATE` doesn't have any condition that would mention `:id_master` or `:id_duplicate`, so it acts on all data 😬 
- The `DELETE` deletes _master's_ archive folder 😬 
- The second update moves duplicate's folders to be owned by master.

I think the correct thing to do would be:
- `UPDATE` threads that are in duplicate's ARCHIVE folder to be in master's archive folder (i.e. modify the first `UPDATE` a bit)
- Update duplicate's folders other than ARCHIVE to be owned by master

There should be no need to `DELETE` anything, as that's handled by cascade when the `message_account` is deleted.

Reply:
Great catches! Did some last minute updates to this and might have been too quick on the trigger 😬 

Updated the PR as per your suggestions.